parsing/schools/jhu/courses.py

```diff
@@ -112,18 +112,29 @@ def _compute_size_enrollment(self, course):
             waitlist = None
         return (section_size, section_enrolment, waitlist)
 
-    def _load_ingestor(self, course, section):
+    def _load_ingestor(self, school, course, section):
+        self.ingestor['sub_school'] = school;
         section_details = section[0]['SectionDetails']
         try:
             num_credits = float(course['Credits'])
         except:
             num_credits = 0
 
-        # Load core course fields
-        self.ingestor['areas'] = filter(lambda a: a != "None",
-                                        course['Areas'].split(','))
-        if course['IsWritingIntensive'] == "Yes":
-            self.ingestor['areas'] += ['Writing Intensive']
+
+        areas = []
+        if school == "Krieger School of Arts and Sciences" or school == "Whiting School of Engineering":
+            for letter in course['Areas']:
+                areas.append(letter.encode('ascii', 'ignore'))
+            # Add specialty areas for computer science department
+            #if course['Department'] == 'EN Computer Science':
+            #    cs_areas_re = r'\bApplications|\bAnalysis|\bSystems|\bGeneral'
+            #    for match in re.findall(cs_areas_re, self.ingestor['description']):
+            #        areas.append(match.encode('ascii', 'ignore'))
+            self.ingestor['areas'] = areas
+        else:
```

Comment:
nitpick but you don't need this else because areas will be empty anyways. Just move the ingest = areas declaration outside of the id